crates/rspack_core/src/compiler/mod.rs

```diff
@@ -421,6 +396,42 @@ impl Compiler {
     Ok(())
   }
 
+  async fn run_clean_options(&mut self) -> Result<()> {
+    let clean_options = &self.options.output.clean;
+    if let CleanOptions::Boolean(false) = clean_options {
+      return Ok(());
+    }
+
+    if self.emitted_asset_versions.is_empty() {
+      self
+        .output_filesystem
+        .remove_dir_all(&self.options.output.path)
+        .await?;
+      return Ok(());
+    }
```

Comment:
When I run `rspack build`, the code reaches this point, and it seems that logic related to `clean` needs to be added here as well.